k8s/kubicle/cloudinit.go

```diff
@@ -0,0 +1,106 @@
+//
+
+package main
+
+const workloadTemplate = `description: "{{.Description}}"
+vm_type: qemu
+fw_type: legacy
+defaults:
+    vcpus: {{.VCPUs}}
+    mem_mb: {{.RAM}}
+cloud_init: "{{.UserDataFile}}"
+disks:
+  - source:
+       service: image
+       id: "{{.ImageUUID}}"
+    size: {{.Disk}}
+    ephemeral: true
+    bootable: true
+`
+
+const udCommonTemplate = `{{- define "PROXIES" -}}
+{{if len .HTTPSProxy }}https_proxy={{.HTTPSProxy}} {{end -}}
+{{if len .HTTPProxy }}http_proxy={{.HTTPProxy}} {{end -}}
+{{end -}}
+---
+#cloud-config
+{{- if len (or $.HTTPProxy $.HTTPSProxy "")}}
+write_files:
+ - content: |
+     [Service]
+     Environment={{if len .HTTPProxy}}"HTTP_PROXY={{$.HTTPProxy}}" {{end}}{{if len .HTTPSProxy}}"HTTPS_PROXY={{.HTTPSProxy}}{{end}}"{{if len .NoProxy}} "NO_PROXY={{.NoProxy}}{{end}}"
+   path: /etc/systemd/system/docker.service.d/http-proxy.conf
+{{- end}}
+
+apt:
+{{- if len $.HTTPProxy }}
+  proxy: "{{$.HTTPProxy}}"
+{{- end}}
+{{- if len $.HTTPSProxy }}
+  https_proxy: "{{$.HTTPSProxy}}"
+{{- end}}
+
+users:
+  - name: {{.User}}
+    gecos: k8s Demo User
+    groups: docker
+    lock-passwd: true
+    shell: /bin/bash
+    sudo: ALL=(ALL) NOPASSWD:ALL
+    ssh-authorized-keys:
+      - {{.PublicKey}}
+
+runcmd:
+ - systemctl restart networking`
+
+const udMasterTemplate = `
+ - echo "export KUBECONFIG=/home/{{.User}}/admin.conf" > /home/{{.User}}/.bash_aliases
+{{- if len .HTTPSProxy }}
+ - echo "export https_proxy={{.HTTPSProxy}}" >> /home/{{.User}}/.bash_aliases
+{{- end}}
+{{- if len .HTTPProxy }}
+ - echo "export http_proxy={{.HTTPProxy}}" >> /home/{{.User}}/.bash_aliases
+{{- end}}
+ - echo "export no_proxy={{if len .NoProxy}}{{.NoProxy}},{{end}}` + "`hostname -i`" + `" >> /home/{{.User}}/.bash_aliases
+ - chown {{.User}}:{{.User}} /home/{{.User}}/.bash_aliases
+ - apt-get update && apt-get install -y apt-transport-https
+ - {{template "PROXIES" .}}curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+ - echo "deb http://apt.kubernetes.io/ kubernetes-xenial main" >/etc/apt/sources.list.d/kubernetes.list
+ - apt-get update
+ - DEBIAN_FRONTEND=noninteractive DEBCONF_NONINTERACTIVE_SEEN=true apt-get install -y docker-engine
+ - DEBIAN_FRONTEND=noninteractive DEBCONF_NONINTERACTIVE_SEEN=true apt-get install -y kubelet kubeadm kubectl kubernetes-cni
+ - {{template "PROXIES" .}}no_proxy=` + "`hostname -i`" + ` kubeadm init --pod-network-cidr 10.244.0.0/16 --token {{.Token}} {{if len .ExternalIP}}--apiserver-cert-extra-sans={{.ExternalIP}}{{end}}
+ - cp /etc/kubernetes/admin.conf /home/{{.User}}/
+ - chown {{.User}}:{{.User}} /home/{{.User}}/admin.conf
+ - {{template "PROXIES" .}}wget https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel-rbac.yml
+ - KUBECONFIG=/home/{{.User}}/admin.conf kubectl create -f kube-flannel-rbac.yml
+ - {{template "PROXIES" .}}wget https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
```

Comment:
We did, but I'll do this in a follow on PR.  kubicle is still functional without this change.